index.js

```diff
@@ -34,7 +35,9 @@ module.exports = {
   create: function(options) {
     var provider;
 
-    if (options.provider) {
+    if (options.provider && _.isFunction(options.provider)) {
```

Comment:
My gut says we don't need a whole dependency for this. Can you change this line to:

```
if (options.provider && typeof options.provider == "function")) {
```